src/components/swap/components/transaction-review/provider.tsx

```diff
@@ -100,30 +103,47 @@ export function useTransactionReview(props: ReviewProps) {
 
   // const { simulateTrade } = useSimulateQuote(quote?.tx ?? null)
 
+  const setLatestTrade = useSetAtom(tradeAtom)
+
   const utm = useUtmParams()
 
   const client = usePublicClient()
   const queryClient = useQueryClient()
   const saveTrade: TradeCallback = useCallback(
     async ({ address, hash, quote }) => {
-      await fetch(`/api/user/trade`, {
+      const trade = mapQuoteToTrade(address, hash, quote, utm)
+
+      setLatestTrade({
+        ...(trade as PostApiV2Trade200),
+        status: 'pending',
+      })
+
+      const postTrade = fetch(`/api/user/trade`, {
         method: 'POST',
-        body: JSON.stringify(mapQuoteToTrade(address, hash, quote, utm)),
+        body: JSON.stringify(trade),
       })
 
-      await client?.waitForTransactionReceipt({
+      const waitForRc = client?.waitForTransactionReceipt({
         hash: hash as `0x${string}`,
         confirmations: 3,
       })
 
+      const [rc, response] = await Promise.all([waitForRc, postTrade])
```

Comment:
Isn't promise.all async? but those calls would have to be done one after the other?

Reply:
ah wait the diff confused me, you got the hash up there.

Reply:
I was basically messing around with sending the save ASAP, so that it goes out immediately and we are not waiting for it, but we actually get it to use later.